### PR TITLE
Add `backend` configuration key to fim integration

### PR DESCRIPTION
**Pull request by Tacklebox**
## Proposed commit message

This adds the configuration key `force_backend` to select an event source for the fim integration on Linux. In order to support including user information on file integrity events across a wider range of linux kernel versions, two new event sources for auditbeat are being developed: `ebpf`, and `kprobes`. In order for a user to be able to select which one they would like to enable a configuration key is being added.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- https://github.com/elastic/integrations/issues/7401

### CODE REVIEW

**Comment by pkoutsovasilis** [comment]
ty @Tacklebox this LGTM, before I press the `Approve` @mmat11 any thoughts?

File: packages/fim/data_stream/event/agent/stream/file_integrity.yml.hbs [hunk 1]
```diff
@@ -5,6 +5,7 @@ paths:
 {{/each}}
 recursive: {{recursive}}
 scan_at_start: {{scan_at_start}}
+force_backend: {{force_backend}}
```
Comment (efd6) [inline packages/fim/data_stream/event/agent/stream/file_integrity.yml.hbs:8]: Does this have a configuration target yet? I don't see one in auditbeat.

File: packages/fim/manifest.yml [hunk 2]
```diff
@@ -72,6 +72,16 @@ vars:
     required: false
     default:
       - sha1
+  - name: force_backend
+    type: text
+    title: File Event Source
+    description: |
+      Forces a particular event source for file events on Linux. `fsnotify` does not provide
+      user information. Supported types are:
+      `ebpf`, `fsnotify`, `kprobe`.
```
Comment (efd6) [inline packages/fim/manifest.yml:81]: Suggest using a select type
```suggestion
    type: select
    title: File Event Source
    description: |
      Forces a particular event source for file events on Linux. `fsnotify` does not provide
      user information.
    options:
     - text: ebpf
       value: ebpf
     - text: fsnotify
       value: fsnotify
     - text: kprobe
       value: kprobe
```

**Review by efd6 — COMMENTED** [review]
Add manifest version bump and changelog entries?

**Comment by mmat11** [comment]
> ty @Tacklebox this LGTM, before I press the `Approve` @mmat11 any thoughts?

LGTM, this should be merged after both the new backends are in auditbeat, right?

**Comment by pkoutsovasilis** [comment]
ty for the review @efd6 and the to the point comments. As @mmat11 mentioned above, this config change captured in the PR is about two new BackEnds for the file_integrity module of auditbeat, and I tend to agree here first merge the respective changes in the beats repo and then merge this one, any thoughts on that @efd6 ?

**Comment by bhapas** [comment]
> ty for the review @efd6 and the to the point comments. As @mmat11 mentioned above, this config change captured in the PR is about two new BackEnds for the file_integrity module of auditbeat, and I tend to agree here first merge the respective changes in the beats repo and then merge this one, any thoughts on that @efd6 ?

@pkoutsovasilis The `beats` changes have to go in first and until `beats` latest is not released this change will not work.

Also, Can you reference the beats issue/PR in this PR.

**Comment by jamiehynds** [comment]
> > ty for the review @efd6 and the to the point comments. As @mmat11 mentioned above, this config change captured in the PR is about two new BackEnds for the file_integrity module of auditbeat, and I tend to agree here first merge the respective changes in the beats repo and then merge this one, any thoughts on that @efd6 ?
> 
> @pkoutsovasilis The `beats` changes have to go in first and until `beats` latest is not released this change will not work.
> 
> Also, Can you reference the beats issue/PR in this PR.

@bhapas Beats PR here: https://github.com/elastic/beats/pull/38199

File: packages/fim/data_stream/event/agent/stream/file_integrity.yml.hbs (hunk 1 above)
Comment (pkoutsovasilis) [reply packages/fim/data_stream/event/agent/stream/file_integrity.yml.hbs]: Hello @efd6, the multiple backends for FIM is now part of auditbeat and [here](https://github.com/elastic/beats/blob/main/auditbeat/module/file_integrity/_meta/config.yml.tmpl#L63) is the configuration target?!

**Review by nfritts — APPROVED** [review]

**Pull request merged**